src/mapml-viewer.js

```diff
@@ -159,26 +173,25 @@ export class MapViewer extends HTMLElement {
       // have a defined width and height.
       var s = window.getComputedStyle(this),
         wpx = s.width, hpx=s.height,
-        w = parseInt(wpx.replace('px','')),
-        h = parseInt(hpx.replace('px',''));
-
-      if (wpx === "" || hpx === "") {
-         return;
-      }
-
-      if (!this.width || this.width !== w) {
-        this._container.style.width = wpx;
-        this.width = w;
-      } else {
-        this._container.style.width = this.width+"px";
-      }
-
-      if (!this.height || this.height !== h) {
-        this._container.style.height = hpx;
-        this.height = h;
-      } else {
-        this._container.style.height = this.height+"px";
-      }
+        w = this.hasAttribute("width") ? this.getAttribute("width") : parseInt(wpx.replace('px','')),
+        h = this.hasAttribute("height") ? this.getAttribute("height") : parseInt(hpx.replace('px',''));
+      // if (!this.width || this.width !== w) {
+      //   this._container.style.width = wpx;
+      // } else {
+      //   this._container.style.width = this.width+"px";
+      // }
+   
+      this._changeWidth(w);
+       
+
```

Comment:
You should remove all commented code so that it does not cause confusion and try to remove unneeded new lines.

Reply:
Also this portion of the code and some other sections seems to be missing from web-map.js, make sure you are making the same changes to both